src/charm.py

```diff
@@ -136,75 +111,22 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         container.replan()
         self._update_app_and_unit_status(ActiveStatus())
 
-    def _database_uri(self) -> typing.Dict[str, str]:
-        """Compute DatabaseURI and return it.
-
-        Returns:
-            DatabaseURI containing details about the data provider integration
-        """
-        db_uris: typing.Dict[str, str] = {}
-
-        # the database_requirer could not be defined
-        # if _database_uri() is called before its initialization
-        if not hasattr(self, "databases") or not self.databases:
-            return db_uris
-
-        for database, db_requires in self.databases.items():
-            relation_data = list(db_requires.fetch_relation_data().values())
-
-            if not relation_data:
-                continue
-
-            # There can be only one database integrated at a time
-            # see: metadata.yaml
-            data = relation_data[0]
-
-            # Check that the relation data is well formed according to the following json_schema:
-            # https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/mysql_client/v0/schemas/provider.json
-            if not all(data.get(key) for key in ("endpoints", "username", "password")):
-                logger.warning("Incorrect relation data from the data provider: %s", data)
-                continue
-
-            database_name = data.get("database", db_requires.database)
-            endpoint = data["endpoints"].split(",")[0]
-            db_uris[database] = (
-                f"{database}://"
-                f"{data['username']}:{data['password']}"
-                f"@{endpoint}/{database_name}"
-            )
-
-        return db_uris
-
-    def _get_flask_env_config(self) -> dict[str, str]:
-        """Return an envConfig with some core configuration.
-
-        Returns:
-            Dictionary with the environment variables for the container.
-        """
-        env_config: typing.Dict[str, str] = {}
-        env_config.update(
-            {
-                f"FLASK_{db_name.upper()}_DB_CONNECT_STRING": db_uri
-                for (db_name, db_uri) in self._database_uri().items()
-            }
-        )
-        env_config.update(self._flask_app.flask_environment)
-        return env_config
-
     def flask_layer(self) -> dict:
         """Generate the pebble layer definition for flask application.
 
         Returns:
             The pebble layer definition for flask application.
         """
+        environment = self._flask_app.flask_environment
+        environment.update(self.databases.get_uris())
```

Comment:
Should we be consistent in whether the environment is returned as an attribute or as a function call?

Reply:
I find it strange that a property could raise an exception

Reply:
So should `.flask_environment` be changed to a function then?

Reply:
Changed to a function